modules/3d/include/opencv2/3d/volume.hpp

```diff
@@ -5,21 +5,57 @@
 #ifndef OPENCV_3D_VOLUME_HPP
 #define OPENCV_3D_VOLUME_HPP
 
+#include "volume_settings.hpp"
+
 #include "opencv2/core/affine.hpp"
 
 namespace cv
 {
+
+
+enum class VolumeType
+{
+    TSDF = 0,
+    HashTSDF = 1,
+    ColorTSDF = 2
+};
+
+
 class CV_EXPORTS_W Volume
+{
+public:
+    Volume();
+    Volume(VolumeType vtype, const VolumeSettings& settings);
+    ~Volume();
+
+    void integrate(OdometryFrame frame, InputArray pose);
+    void integrate(InputArray frame, InputArray pose);
+    void raycast(InputArray cameraPose, int height, int width, OutputArray points, OutputArray normals) const;
```

Comment:
Maybe to add `raycast(InputArray cameraPose, int height, int width, OdometryFrame& outFrame)` to correspond to `integrate()`?

Reply:
done, volume works with OdometryFrame